helix-view/src/editor.rs

```diff
@@ -1259,6 +1262,21 @@ impl Editor {
         self.tree.is_empty()
     }
 
+    /// Gets the count pair (warnings, error) of warnings, error on workspace.
+    pub fn workspace_diagnostic_count(&self) -> (i32, i32) {
+        self.diagnostics
+            .values()
+            .flatten()
+            .fold((0, 0), |mut counts, diag| {
+                match diag.severity {
+                    Some(DiagnosticSeverity::WARNING) => counts.0 += 1,
+                    Some(DiagnosticSeverity::ERROR) | None => counts.1 += 1,
+                    _ => {}
+                }
+                counts
+            })
+    }
```

Comment:
Especially since these (`workspace_diagnostic_count` and `diagnostics_count`) filter on severity I think these seem specific to the statusline element and should be inlined there: I don't think these functions would be re-used outside of this use-case